### DISCUSSION TITLE
Plot  contours in IPF using MTeX

### DISCUSSION THREAD

**Discussion by XiaoYinZ** [mtex-toolbox/mtex#1481] [Ask Anything] [answered]
Hi,

I have some crystal directions by Euler angle, and a corresponding scalar value associated with each direction.  Now i want to visualize this data as filled contours in IPF map.When i use"plotIPDF(ori,value,vector3d.X);colorbar", figure shows points and corresponding value. But when I use "plotIPDF(ori,value,vector3d.X,'contourf');colorbar'" to finish it, there are no lines. 

So, how can i visualize corresponding scalar value as filled contours?
Thanks a lot!

fname=load("1.text")
cs = crystalSymmetry('cubic');
ss = specimenSymmetry('orthorhombic');
ori = orientation.byEuler(fname(:,1)*degree,fname(:,2)*degree,fname(:,3)*degree,cs,ss)
data = [12.22664669
12.32936438
18.18337354
13.26040428
11.95305619
10.22900743
14.13820103
15.31273635
11.47217949
11.43377074
11.62181178
13.94393095
10.85557681
10.9166289
15.18632351
18.2196188
11.09013849
10.28189946
13.7318742
18.37844515
10.77109258
17.44239645
11.96821648
18.82527321
12.27323677
18.74650434
12.52206564
12.8531433
18.32847627
19.79586362
14.03159428
17.20520886
19.1943093
15.04827982
12.36744588
19.93935317
13.30106768
14.09361146
16.45669415
15.68561108
15.21907946
16.50683721
11.76867246
17.70067619
15.63542972
18.48086627
16.45823422
11.71214837
11.57784137
11.09452125
12.8088354
11.25159425
18.45429273
12.31791399
11.14031559
16.43808797
12.64502812
12.32936438
18.18337354
13.26040428
11.95305619
10.22900743
14.13820103
15.31273635
11.47217949
11.43377074
11.62181178
13.94393095
10.85557681
10.9166289
15.18632351
18.2196188
11.09013849
10.28189946
13.7318742
18.37844515
10.77109258
17.44239645
11.96821648
18.82527321
12.27323677
18.74650434
12.52206564
12.8531433
18.32847627
19.79586362
14.03159428
17.20520886
19.1943093
15.04827982
12.36744588
19.93935317
13.30106768
14.09361146
16.45669415
15.68561108
15.21907946
16.50683721
11.76867246
17.70067619
15.63542972
18.48086627
16.45823422
11.71214837
11.57784137
11.09452125
12.8088354
11.25159425
18.45429273
12.31791399
11.14031559
16.43808797
12.64502812
12.32936438
18.18337354
13.26040428
11.95305619
10.22900743
14.13820103
15.31273635
11.47217949
11.43377074
11.62181178
13.94393095
]
%plotIPDF(ori,value,vector3d.X,'contourf')   %%error
plotIPDF(ori,data,vector3d.X)

colorbar

**Comment by kilir** [comment]
Hi,
so how does your plotlook like instead?
Cheers,
Rüdiger

**Comment by XiaoYinZ** [comment]
Hi,
my figure look like as follows:
![This is an image](https://github.com/XiaoYinZ/figure/blob/main/1.jpg)
plotIPDF(ori,data,vector3d.X)
![This is an image](https://github.com/XiaoYinZ/figure/blob/main/2.jpg)
plotIPDF(ori,data,vector3d.X,'contourf')

and i want it looks like:
![This is an image](https://github.com/XiaoYinZ/figure/blob/main/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20220707140334.png)

How can i map the first graph to the contour map as the third graph?

Thank you!

**Reply by kilir** [reply]
Omitting the specimen symmetry should do the trick.
Cheers,
Rüdiger

**Reply by XiaoYinZ** [reply]
Thank you for your comments, It is a pity that when omitting the specimen symmetry, error still reported after attempt.


This is the file "1.text":

  108.25  124.83   44.01 
 -119.25   52.33   46.39 
  167.46   51.72   45.03 
  144.48   62.07   41.23 
  149.21   79.59    8.54 
   27.75    1.78 -134.35 
   17.90   57.81  -42.56 
 -163.39  169.93  -19.34 
  -47.96   55.14  134.71 
  -68.34  126.33 -135.65 
  -11.33  145.81  -46.64 
   62.14  178.16  105.41 
   42.41   55.74   44.51 
   20.24  124.08  138.30 
   47.01  124.30  -42.15 
  -90.60   86.40  177.53 
  112.94    3.53 -169.41 
   82.45   56.49   35.84 
   73.29   55.04 -136.78 
  -51.28   42.05  139.64 
 -146.32  124.89  -43.06 
   -7.11  131.34 -131.69 
  -76.07    4.12   46.04 
  179.02   52.02 -135.11 
  143.37   67.85   59.87 
  116.36  122.93  -46.88 
  -63.44   90.44  -88.31 
   -4.13   55.80  148.62 
  -86.14   90.01  -92.65 
  -38.84  125.61  -43.85 
  132.34   64.35   59.23 
   18.10   53.89  -44.40 
 -175.99  125.21  133.28 
  -77.58   57.83 -128.73 
  -41.54  126.20 -134.57 
   36.43   55.08 -133.15 
   21.65  126.80 -135.29 
 -115.96   57.66 -156.86 
 -112.39  122.52  131.10 
   23.58   83.91   86.20 
 -131.30  126.06   46.43 
  114.70   87.42   92.04 
  157.21   57.63 -141.38 
  -59.76   54.20  134.75 
  174.44   55.30 -131.72 
 -152.74   91.93 -141.17 
 -105.67   52.50   37.62 
  131.38   56.62  134.47 
  -51.96   53.35 -139.19 
  -84.52   49.85 -140.72 
 -102.04   46.56  -46.53 
  -73.95  117.00  146.93 
   37.24  124.51 -135.02 
  164.89  134.89  -44.49 
   70.91   54.88 -135.12 
  -99.79  124.31  -41.09 
   86.12  124.37  -47.40 
  -22.29   53.01   42.73 
  -62.04    1.33  -24.65 
   46.57   89.80   95.24 
  157.34    2.49   -9.82 
  -23.69  123.92  137.27 
 -104.24   56.10 -134.84 
  169.84  176.69  -62.16 
  -52.67   56.74 -135.03 
  165.27   55.17 -135.48 
   20.06  127.67 -135.56 
   40.97  125.20 -135.38 
 -106.02   41.38   58.83 
 -134.68   55.55 -136.06 
   15.03   57.78   40.67 
   -3.05  103.12  167.37 
  115.59   21.89  136.73 
 -123.52   74.21  -55.95 
   63.20  170.39  123.69 
  -59.60  176.95   23.34 
    7.71  128.81 -136.79 
  163.24   53.57   45.14 
  130.85   63.26   63.90 
    4.44  127.20   44.98 
 -176.72   55.77  139.87 
   73.51    6.74  170.10 
  -26.30  131.81  133.53 
  -43.89  122.55 -141.38 
   42.03    1.01   46.03 
  154.28   59.83 -146.66 
  178.32   62.05  137.03 
   55.48   90.33   95.35 
   11.55  124.91   46.05 
   14.70   56.12  -45.77 
  136.49   24.62   27.50 
   67.97   43.51   42.68 
   94.49   85.03  175.30 
   30.72  125.97 -134.61 
  -51.94   92.31   -1.09 
 -173.64  127.55 -134.82 
  122.68  126.36  134.33 
   65.20   57.13  -47.91 
  100.61  124.39  -45.87 
 -165.80   75.69   72.09 
   75.96   55.78  -44.43 
  179.95   88.20   88.31 
  127.49   54.78   43.48 
  170.28  126.52  133.93 
   40.29   91.25   89.67 
  124.30  100.54  -48.95 
   23.03   15.54   28.82 
  -35.60   54.41  135.11 
    0.07  124.98 -133.92 
   -8.91   58.83 -129.60 
   16.25   52.38 -134.01 
   54.71  124.62   47.42 
  -80.82  124.79  134.80 
 -143.72   55.39 -136.56 
  109.87  120.75 -125.90 
  126.58  125.52  -44.23 
   71.91   55.46   43.68 
  -70.64   53.55  -45.54 
 -140.10  123.95   40.54 
   25.80  141.45  -17.13 
   28.91   15.73  -46.62 
   12.79   91.62   90.16 
   67.09   55.46  170.26 
  -89.53   54.99  136.09

**Comment by kilir** [comment, marked as answer]
If you get an error message, it might help pasting it here. If you get an error about dimension mismatch ultimately in `S2FunHarmonic/approximation.m`, try if this here c52d339d6e30c4d7a3c51ebd6e0faee9c7d369cc solves it for you.
Cheers,
Rüdiger

**Reply by XiaoYinZ** [reply]
Yes, it works.
Thank you very much for your kind answer.
